app/processors/message_processor.py

```diff
@@ -13,17 +15,26 @@ def __init__(self):
         self.ftp = SDXFTP(settings.FTP_HOST, settings.FTP_USER, settings.FTP_PASS)
 
     def process(self, msg, tx_id):
+        id_tag = json.loads(msg)
         if tx_id is None:
-            tx_id = msg
+            tx_id = id_tag.get('tx_id')
```

Comment:
why does this need to change? 

The tx_id is passed as a parameter so not to affect existing functionality we probably should continue to use that

Reply:
like I mentioned in the sdx-collect review, I think it may be better to add a header to message rather than modifying the message payload itself.